gdal2mbtiles/vips.py

```diff
@@ -307,8 +308,15 @@ def _scale(self, xscale, yscale, interpolate):
         # The corners of output.img are located at:
         #     (-.5,-.5), (-.5,M-.5), (N-.5,-.5) and (N-.5,M-.5).
 
-        output_width = int(ceil(self.image.width * xscale))
-        output_height = int(ceil(self.image.height * yscale))
+        if output_size is None:
+            if XY(x=xscale, y=yscale) > XY(x=1.0, y=1.0):
+                output_width = int(ceil(self.image.width * xscale))
+                output_height = int(ceil(self.image.height * yscale))
+            else:
+                output_width = int(floor(self.image.width * xscale))
+                output_height = int(floor(self.image.height * yscale))
+        else:
+            output_width, output_height = output_size
 
```

Comment:
@mghughes I'm concerned that this will reintroduce the bug fixed by https://github.com/ecometrica/gdal2mbtiles/commit/4dcd9f1b613584ab6bdfc3631a44aada259af230 (and maybe explains the reason for the missing test file and commented out test for downsampling to native resolutions).  Should I revert it?

Reply:
Right, yeah, this might be problematic. Probably best to revert it.

Reply:
I'll put it back but try to update the test/code comments so it doesn't confuse future me again